### PR TITLE
Bot buy can fail

### PR DESCRIPTION
**Pull request by bmorel**
Allow BT to handle if buying a weapon or upgrade failed. This should never happen if nothing specific was requested (`action equip` or `action evolve`) but can happen if a specific equipment or class is asked for which is not unlocked yet, forbidden to buy (by either bot specific cvars or general disabling of equipment) or bot does not have the ¤ for, resulting for humans in possibly selling the weapon and roaming around with a blaster (no ckit nor smg).

This PR does not fix any situation, it only allows things to be saved.

### CODE REVIEW

**Comment by bmorel** [comment]
Code is extracted from #1654

File: src/sgame/sg_bot_util.cpp [hunk 1]
```diff
@@ -1986,111 +1986,112 @@ bool BotEvolveToClass( gentity_t *ent, class_t newClass )
...
 
-		//can afford this?
-		if ( BG_Weapon( weapon )->price > ( short )self->client->pers.credit )
-		{
-			return;
-		}
+	//already got this?
+	if ( BG_InventoryContainsWeapon( weapon, self->client->ps.stats ) )
+	{
+		return false;
```
Comment (slipher) [inline src/sgame/sg_bot_util.cpp:1999]: I wouldn't call that a failure

File: src/sgame/sg_bot_util.cpp [hunk 2]
```diff
@@ -1986,111 +1986,112 @@ bool BotEvolveToClass( gentity_t *ent, class_t newClass )
...
 	{
-		return;
+		return false;
 	}
 
 	//already got this?
 	if ( BG_InventoryContainsUpgrade( upgrade, self->client->ps.stats ) )
 	{
-		return;
+		return false;
```
Comment (slipher) [inline src/sgame/sg_bot_util.cpp:2064]: I wouldn't call this a failure

**Comment by necessarily-equal** [comment]
> This should never happen if nothing specific was requested (action equip or action evolve)

Well, it can :P — that's among the bugs I'll have to fix before I can make a tutorial release. See https://github.com/Unvanquished/Unvanquished/issues/1586#issuecomment-968281421 for a description and a reproducer.

**Comment by bmorel** [comment]
> Well, it can :P — that's among the bugs I'll have to fix before I can make a tutorial release. See #1586 (comment) for a description and a reproducer.

One more reason to make BT able to handle the situation, then.

File: src/sgame/sg_bot_util.cpp (hunk 1 above)
Comment (bmorel) [reply src/sgame/sg_bot_util.cpp:1999]: Agreed.

File: src/sgame/sg_bot_util.cpp (hunk 2 above)
Comment (bmorel) [reply src/sgame/sg_bot_util.cpp:2064]: Agreed.

**Review by necessarily-equal — APPROVED** [review]
From a quick look, LGTM

**Pull request merged**